cai2hcl/common.go

```diff
@@ -3,23 +3,22 @@ package cai2hcl
 import (
 	"github.com/GoogleCloudPlatform/terraform-google-conversion/v2/caiasset"
 
-	tfschema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/zclconf/go-cty/cty"
 )
 
-// ConvertFunc converts an Asset type into a map[string]interface{} struct.
-type ConvertFunc func(asset *caiasset.Asset) (string, map[string]interface{}, error)
-
-// Converter for individual resource.
-type Converter struct {
-	TFResourceName string
-	Convert        ConvertFunc
-	Resource       *tfschema.Resource
+// Converter interface for resources.
+type Converter interface {
+	// Convert converts an Asset into ID, and cty.Value.
+	Convert(asset ...*caiasset.Asset) (id string, data cty.Value, iam cty.Value, err error)
+	// TFResourceType returns terraform resource type.
+	TFResourceType() string
 }
 
 // Converters returns a map of resource converters.
 // The map key is the CAI Asset type.
-func Converters() map[string]*Converter {
-	return map[string]*Converter{
+func Converters() map[string]Converter {
```

Comment:
This would also make supporting IAM simpler - you would be able to have one converter for the core resource and a separate converter for the IAM policy.

Reply:
Did not use []Converter. It changed to use map[string]string -> key = asset type, value = converter name (just the terraform resource name). Then after grouping by converter name, found the converter from map[string]Converter -> key = converter name, value = Converter, it's to avoid initiating multiple converter instances, and not using converter address as map key.